test/share-socket.js

```diff
@@ -0,0 +1,423 @@
+    coap.request('coap://localhost:'+port + '/abcd/ef/gh/?foo=bar&beep=bop').end()
+    server.on('request', function(req) {
+      expect(req.url).to.eql('/abcd/ef/gh?foo=bar&beep=bop')
+      setImmediate(done)
+    })
+  })
+
+  it('should return code 2.05 by default', function(done) {
+    var req = coap.request('coap://localhost:'+port + '/abcd/ef/gh/?foo=bar&beep=bop').end()
+    req.on('response', function(res) {
+      expect(res.code).to.eql('2.05')
+      setImmediate(done)
+    })
+
+    server.on('request', function(req, res) {
+      res.end('hello')
+    })
+  })
+
+  it('should return code using res.code attribute', function(done) {
+    coap
+      .request('coap://localhost:'+port)
+      .on('response', function(res) {
+        expect(res.code).to.eql('4.04')
+        setImmediate(done)
+      })
+      .end()
+
+    server.on('request', function(req, res) {
+      res.code = '4.04'
+      res.end('hello')
+    })
+  })
+
+  it('should return code using res.statusCode attribute', function(done) {
+    coap
+      .request('coap://localhost:'+port)
+      .on('response', function(res) {
+        expect(res.code).to.eql('4.04')
+        setImmediate(done)
+      })
+      .end()
+
+    server.on('request', function(req, res) {
+      res.statusCode = '4.04'
+      res.end('hello')
+    })
+  })
+
+  it('should support observing', function(done) {
+    var req = coap.request({
+        port: port
+      , observe: true
+    }).end()
+
+    req.on('response', function(res) {
+      res.once('data', function(data) {
+        expect(data.toString()).to.eql('hello')
+        res.once('data', function(data) {
+          expect(data.toString()).to.eql('world')
+          done()
+        })
+      })
+    })
+
+    server.on('request', function(req, res) {
+      res.write('hello')
+      res.end('world')
+    })
+  })
+
+  it('should support a 4.04 observe request', function(done) {
+    var req = coap.request({
+        port: port
+      , observe: true
+    }).end()
+
+    req.on('response', function(res) {
+      expect(res.code).to.eql('4.04')
+      done()
+    })
+
+    server.on('request', function(req, res) {
+      res.statusCode = '4.04'
+      res.end()
+    })
+  })
+
+  it('should support a 4.04 observe request and emit an end event in the response', function(done) {
+    var req = coap.request({
+        port: port
+      , observe: true
+    }).end()
+
+    req.on('response', function(res) {
+      expect(res.code).to.eql('4.04')
+      res.on('end', done)
+      res.resume()
+    })
+
+    server.on('request', function(req, res) {
+      res.statusCode = '4.04'
+      res.end()
+    })
+  })
+
+  describe('formats', function() {
+    var formats = [ 'text/plain', 'application/link-format',
+      'application/xml', 'application/octet-stream',
+      'application/exi', 'application/json', 'application/cbor' ]
+
+    ;['Accept', 'Content-Format'].forEach(function(option) {
```

Comment:
Ah sorry, I was too hasty to ctrl+f ; and comment, without looking at it closer :)